### PR TITLE
Filter for billings done

### PR DESCRIPTION
**Pull request by ajgo**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,6 +66,19 @@
       <artifactId>mapstruct-jdk8</artifactId>
       <version>${org.mapstruct.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.poi</groupId>
+      <artifactId>poi</artifactId>
+      <version>4.0.1</version>
```
Comment (adamzareba) [inline pom.xml]: move pls version to properties section

File: pom.xml [hunk 2]
```diff
@@ -66,6 +66,19 @@
...
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/org.apache.poi/poi-ooxml -->
+    <dependency>
+      <groupId>org.apache.poi</groupId>
+      <artifactId>poi-ooxml</artifactId>
+      <version>4.0.1</version>
```
Comment (adamzareba) [inline pom.xml]: move pls version to properties section

File: src/main/java/com/example/event_manager/service/TaskStatusServiceImpl.java [hunk 3]
```diff
@@ -32,7 +31,7 @@ public TaskStatusForm saveAndReturn(final TaskStatusForm taskStatus) {
   @Override
   public List<TaskStatusForm> saveAndReturnList(List<TaskStatusForm> taskStatuses) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/TaskStatusServiceImpl.java]: final param

File: src/main/java/com/example/event_manager/service/EventServiceImpl.java [hunk 4]
```diff
@@ -152,6 +147,11 @@ public AllEventsForm searchByNamePlaceTopic(final String s, final Pageable pagin
         .build();
   }
 
+  @Override
+  public BillingRaportSchema generateBillingRaportSchemaForEvent(Long id) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/EventServiceImpl.java]: remove it if unused
return NotImplementedException exception if implementation will be provided in future

Comment (adamzareba) [reply src/main/java/com/example/event_manager/service/EventServiceImpl.java]: + final param

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 5]
```diff
@@ -28,87 +28,89 @@ Licensed to the Apache Software Foundation (ASF) under one
...
-            System.exit(0);
-        } catch (Throwable e) {
-            System.out.println("- Error downloading");
-            e.printStackTrace();
-            System.exit(1);
+          if (mavenWrapperPropertyFileInputStream != null) {
+            mavenWrapperPropertyFileInputStream.close();
+          }
+        } catch (IOException e) {
+          // Ignore ...
```
Comment (adamzareba) [inline .mvn/wrapper/MavenWrapperDownloader.java]: why ignore?

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+@AllArgsConstructor
+public class BillingController {
+
+  private final EventService eventService;
+  private final BillingRaportService billingRaportService;
+  private final PersonService personService;
+  private final BillingService billingService;
+
+  @GetMapping("/")
+  public void str() {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: not meaningful name

Comment (adamzareba) [reply src/main/java/com/example/event_manager/controller/BillingController.java]: what's the goal of this method?

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    BillingForm billing = BillingForm.builder().title("Title 1").deleted(false)
+        .dateOfEdition(LocalDateTime.now()).dateOfCreation(LocalDateTime.now())
+        .money(BigDecimal.valueOf(200)).personAssigned(personService.findById(1L)).billingType(
+            BillingType.INCOMING).confirmed(true).build();
+    billingService.save(billing);
+  }
+
+  @GetMapping("/details/{id}")
+  public String billingDetails(Model model, @PathVariable Long id) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: finals

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    model.addAttribute("accountForm", accountForm);
+    model.addAttribute("persons", personService.findAll());
+    model.addAttribute("billingFilter", billingFilter);
+
+    return "billingsDetails";
+  }
+
+  @GetMapping("/details/delete")
+  public String deleteBillingFromEvent(@RequestParam("eventId") Long eventId,
+      @RequestParam("billingId") Long billingId) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: finals

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+    return "details/delete";
+  }
+
+  @GetMapping("/details/raport")
+  public ResponseEntity<byte[]> billings(@RequestParam final Long id)
+      throws IOException {
+    HttpHeaders headers = new HttpHeaders();
+    EventForm ef = eventService.eventFormById(id);
+    headers.add("Content-Disposition", "attachment; filename=billings.xlsx");
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: use org.springframework.http.HttpHeaders.CONTENT_DISPOSITION

inject billings.xlsx via property value and @Value

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    final byte[] excelInByteArray = billingRaportService.generateSheetForBillings(ef.getBillings());
+    return ResponseEntity.ok().headers(headers).body(excelInByteArray);
+  }
+
+  @PostMapping("/details/raport/filter")
+  public ResponseEntity<byte[]> getFilterObject(Model model,
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: finals

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+      @ModelAttribute("billingFilter") BillingFilter billingFilter)
+      throws IOException {
+    System.out.println(billingFilter.getPersonName());
+
+    EventForm ef = eventService.eventFormById(1L);
+    List<BillingForm> filteredBillings = billingRaportService
+        .filterBillingFormList(ef.getBillings(), billingFilter);
+    HttpHeaders headers = new HttpHeaders();
+    headers.add("Content-Disposition", "attachment; filename=billings.xlsx");
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: use org.springframework.http.HttpHeaders.CONTENT_DISPOSITION

inject billings.xlsx via property value and @value

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 12]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
       byteArrayOutputStream.close();
     }
   }
+
+  @Override
+  public List<BillingForm> filterBillingFormList(List<BillingForm> oldBillingForm,
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: finals

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 13]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
...
+      BillingFilter billingFilter) {
+    List<BillingForm> filtered = oldBillingForm
+        .stream()
+        .filter(bf -> {
+          if (billingFilter.getBillingType().equals("all")) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: more secure approach is to put contant value at left side during comparison, like:
"all".equals(billingFilter.getBillingType())
you can avoid NPE then

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 14]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
...
+            return true;
+          } else if (BillingType.valueOf(billingFilter.getBillingType()) == bf.getBillingType()
+              && BillingType.valueOf(billingFilter.getBillingType()) == BillingType.INCOMING) {
+            return true;
+          } else {
+            return BillingType.valueOf(billingFilter.getBillingType()) == bf.getBillingType()
+                && BillingType.valueOf(billingFilter.getBillingType()) == BillingType.OUTLAY;
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: more secure approach is to put contant value at left side during comparison,

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 15]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
...
+        .filter(bf -> bf.getPersonAssigned().getName().equals(billingFilter.getPersonName()))
+        .filter(bf -> bf.isConfirmed() == billingFilter.isConfirmed())
+        .filter(bf -> bf.getDateOfCreation().isAfter(billingFilter.getDateCreationStart()) && bf
+            .getDateOfCreation().isBefore(billingFilter.getDateCreationEnd()))
+        .collect(Collectors.toList());
+    return filtered;
+  }
+
+  @Override
+  public byte[] generateSheetForBillings(List<BillingForm> billingForms) throws IOException {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: final

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 16]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
...
+    XSSFWorkbook xssfWorkbook = new XSSFWorkbook();
+    XSSFSheet sheet = xssfWorkbook.createSheet("BillingsList");
+    Row row = sheet.createRow(0);
+    headersForSheet(row);
+    int rowCount = 1;
+    for (BillingForm billingForm : billingForms) {
+      row = sheet.createRow(rowCount++);
+      Cell cell = row.createCell(0);
+      cell.setCellValue(billingForm.getTitle());
+      cell = row.createCell(1);
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: pls avoid magic numbers like 1,2,3...
create private static final int fields and name it like for example
private static final int NAME_IDX = 1;
private static final int DESCRIPTION_IDX = 2;
and then use it in method code

File: src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java [hunk 17]
```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
...
+    }
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    xssfWorkbook.write(byteArrayOutputStream);
+    return byteArrayOutputStream.toByteArray();
+  }
+
+  private void headersForSheet(Row row) {
+    Cell cell = row.createCell(0);
+    cell.setCellValue("Title");
+    cell = row.createCell(1);
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java]: magic numbers, look at comment above pls

File: src/main/java/com/example/event_manager/service/EventServiceImpl.java [hunk 18]
```diff
@@ -170,15 +174,7 @@ public AllEventsForm getPartition(
...
-    final EventWithBillingForm eventWithBillingForm =
-            eventWithBillingMapper.mapToDTO(repo.findById(id).get());
-    final List<BillingForm> listOfBilling = eventWithBillingForm.getBillings();
-    final BillingsSummary bs = new BillingsSummary(listOfBilling);
-    final BillingRaportSchema brs = new BillingRaportSchema();
-    brs.setEventName(eventWithBillingForm.getName());
-    brs.setBillings(listOfBilling);
-    brs.setBillingsSummary(bs);
-    return brs;
+    return null;
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/service/EventServiceImpl.java]: method do nothing, remove it pls

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 19]
```diff
@@ -28,87 +28,89 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 public class MavenWrapperDownloader {
```
Comment (adamzareba) [inline .mvn/wrapper/MavenWrapperDownloader.java]: revert changes (or gitignore), if it's in repo already then I would recommend revert changes as it is in master branch

File: src/main/java/com/example/event_manager/controller/BillingController.java [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+    model.addAttribute("accountForm", accountForm);
+    model.addAttribute("persons", personService.findAll());
+    model.addAttribute("billingFilter", billingFilter);
+
+    return "billingsDetails";
+  }
+
+  @GetMapping("/details/delete")
+  public String deleteBillingFromEvent(@RequestParam("eventId") final Long eventId,
+      @RequestParam("billingId") Long billingId) {
```
Comment (adamzareba) [inline src/main/java/com/example/event_manager/controller/BillingController.java]: Long billingId should be final

File: src/main/java/com/example/event_manager/controller/BillingController.java (hunk 9 above)
Comment (ajgo) [reply src/main/java/com/example/event_manager/controller/BillingController.java]: Billings.xlsx isn't a file that I store on server so use @Value?
or private final String  = "Billings.xlsx will be okay?

Comment (ajgo) [reply src/main/java/com/example/event_manager/controller/BillingController.java]: It's only name for file that will be returned from controller

Comment (adamzareba) [reply src/main/java/com/example/event_manager/controller/BillingController.java]: inject is as a String from property file, for example
@Value("${report.filename}")
private String filename;

**Pull request merged**